db/view/view_update_generator.hh

```diff
@@ -42,9 +42,10 @@ class view_update_generator {
     struct sstable_with_table {
         sstables::shared_sstable sst;
         lw_shared_ptr<table> t;
-        sstable_with_table(sstables::shared_sstable sst, lw_shared_ptr<table> t) : sst(sst), t(t) { }
+        sstable_with_table(sstables::shared_sstable sst, lw_shared_ptr<table> t) : sst(std::move(sst)), t(std::move(t)) { }
     };
     std::deque<sstable_with_table> _sstables_with_tables;
+    std::unordered_map<lw_shared_ptr<table>, std::vector<sstables::shared_sstable>> _sstables_to_move;
```

Comment:
Is there a reason this one is a map, instead of simply being a deque, just like `_sstables_with_tables`? It seems like it's only ever used to iterate over all of its elements, and deque looks like a more natural structure for that access pattern (no per-item allocations, less overhead, and so on).

Reply:
@psarna it is a map since it is 2-dimensional. I keep a vector of sstables per table (aka column_family) and then call t->move_sstables_from_staging on that vector.  The reason is that they are all stored in the same staging directory so we can sync it and its parent at the end after moving all sstables out of it, rather than syncing at random for each individual sstable.